app/vminsert/netstorage/netstorage.go

```diff
@@ -398,7 +403,8 @@ func (sn *storageNode) dial() (*handshake.BufferedConn, error) {
 	if *disableRPCCompression {
 		compressionLevel = 0
 	}
-	bc, err := handshake.VMInsertClient(c, compressionLevel)
+	bc, id, err := handshake.VMInsertClient(c, compressionLevel)
+	sn.id.CompareAndSwap(0, id)
```

Comment:
shall we always compare with 0? Or shall we compare with previously set id?

Reply:
The idea of comparing to `0` is to assign value only once.
With the latest version there is no fallback to another ID value, so the only path to assign and ID is to get it from the handshake.